internal/controlplane/server_test.go

```
@@ -80,6 +80,35 @@ func TestServerProtectsSubmissionAndWorkerAPIs(t *testing.T) {
+		want int
+	}{
+		{name: "IPv4 with port", host: "127.0.0.1:7331", want: http.StatusOK},
+		{name: "IPv6 with port", host: "[::1]:7331", want: http.StatusOK},
+		{name: "localhost", host: "LOCALHOST:7331", want: http.StatusOK},
+		{name: "DNS rebinding host", host: "attacker.example:7331", want: http.StatusForbidden},
+		{name: "non-loopback IP", host: "192.0.2.1:7331", want: http.StatusForbidden},
+		{name: "invalid port", host: "localhost:not-a-port", want: http.StatusForbidden},
+		{name: "malformed host", host: "localhost:invalid:7331", want: http.StatusForbidden},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			request := httptest.NewRequest(http.MethodGet, "/api/v1/status", nil)
```

golangci-lint (2.13.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)